modules/rte_mis_core/src/Plugin/Block/RoleBasedDetailsBlock.php

```diff
@@ -0,0 +1,599 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\rte_mis_core\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a role based details block block.
+ *
+ * @Block(
+ *   id = "rte_mis_core_role_based_details_block",
+ *   admin_label = @Translation("Role Based Details Block"),
+ *   category = @Translation("Custom"),
+ * )
+ */
+final class RoleBasedDetailsBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The current user service.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs the plugin instance.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    EntityTypeManagerInterface $entityTypeManager,
+    AccountProxyInterface $currentUser,
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->currentUser = $currentUser;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition): self {
+    return new self(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('current_user'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // Check the roles of the current user.
+    $roles = $this->currentUser->getRoles();
+
+    // Initialize content variables.
+    $heading = $this->t('Dashboard');
+    $content = [];
+
+    // Check roles and set content accordingly.
+    if (in_array('state_admin', $roles)) {
+      $heading = $this->t('State Admin Dashboard');
+      $content = $this->getStateAdminContent();
+    }
+    elseif (in_array('district_admin', $roles)) {
+      $heading = $this->t('District Admin Dashboard');
+      $content = $this->getDistrictAdminContent();
+    }
+    elseif (in_array('block_admin', $roles)) {
+      $heading = $this->t('Block Admin Dashboard');
+      $content = $this->getBlockAdminContent();
+    }
+
+    // Build renderable array.
+    $build = [
+      '#theme' => 'role_based_details_block',
+      '#heading' => $heading,
+      '#content' => $content,
+    ];
```

Comment:
No cache Tags, cache contexts are required for this block?

Reply:
yes, we should add caching to it.